docs/blogs/IntroGatewayAPI/GettingCreative.md

```diff
@@ -0,0 +1,295 @@
+match on the `accept-encoding` header in order to use a different backend for clients requesting HTML and clients requesting
+json. Similarly, `queryParams` can match on arbitrary URL query parameters.
+
+The following example shows header and URL query parameter matching. First, we set a match rule using `queryParams` that
+if the `app` query parameter is set to `foo`, the request should be forwarded to the `foo-app` application; and second, we
+set a match rule using headers that if the `x-app` header is set to `bar`, the request should be forwarded to the `bar-app`
+application:
+```yaml
+$ oc apply -f - <<EOF
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http
+  namespace: demo-app
+spec:
+  parentRefs:
+  - name: gateway
+    namespace: demo-gateway
+  hostnames: ["http.example.com"]
+  rules:
+  - matches:
+    - queryParams: #[1]
+      - name: app
+        value: foo
+    backendRefs:
+    - name: foo-app
+      port: 8080
+  - matches:
+    - headers: #[2]
+      - name: x-app
+        value: bar 
+    backendRefs:
+    - name: bar-app
+      port: 8080
+EOF
+
+```
+[1] `queryParams`: if the `app` query parameter is set to `foo`, the request should be forwarded to the `foo-app` application.\
+[2] `headers`: if the `x-app` header is set to `bar`, the request should be forwarded to the `bar-app` application.
+
+Now let’s test query parameter matching:
+```shell
+$ curl -sH "Host: http.example.com" $INGRESS_HOST?app=foo
+Welcome to the foo application!
+```
+
+Next, let's test header matching:
+```shell
+$ curl -sH "Host: http.example.com" -H "x-app: bar" $INGRESS_HOST
+Welcome to the bar application!
+```
+
+You’ve successfully configured Gateway API to match traffic based on path, query parameters, and headers.
+
+## TLS Termination with Gateway API
+In OpenShift, we have the concept of [termination types](https://docs.openshift.com/container-platform/4.11/networking/routes/secured-routes.html) 
+to achieve different forms of secured routing. In this example, we will explore how to create an edge terminated route with Gateway API.
+
+An edge route in OpenShift terminates TLS encryption at the Ingress Controller before forwarding traffic to the
+destination pod:
+```yaml
+apiVersion: v1
+kind: Route
+metadata:
+  name: http-foo
+  namespace: demo-app
+spec:
+  host: www.edge.example.com
+  to:
+    kind: Service
+    name: foo-app
+  port:
+    targetPort: 8080
+tls:
+  termination: edge
+  insecureEdgeTerminationPolicy: Redirect
+  certificate: <CERT>
+  key: <KEY>
+path: /foo
+```
+In Gateway API, we can translate this concept to terminating the TLS encryption at Gateway before forwarding traffic to
+the HTTPRoute.
+
+First, we will need a certificate for configuring the TLS termination. Acquire a certificate with the common name that
+you want to use from your CA.  
+
+For testing purposes, you can create your own CA and serving certificate as follows:
+```shell
```

Comment:
Using ` ```shell ` might cause some funny highlighting because this is terminal output, not shell script.

Reply:
I made them all shell unless they are yaml, but I don't think it will matter for the blog.  I will change this one.